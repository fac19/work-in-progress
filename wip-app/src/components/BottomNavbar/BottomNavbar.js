import React from "react";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import AppsIcon from "@material-ui/icons/Apps";
import PersonIcon from "@material-ui/icons/Person";
import NotificationsIcon from "@material-ui/icons/Notifications";
import ExploreIcon from "@material-ui/icons/Explore";

import {
  BottomNavbarStyle,
  BottomNavbarUlStyle,
  BottomNavbarLiStyle,
} from "./BottomNavbar.style";

const useStyles = makeStyles({
  links: {
    color: "black",
    textDecoration: "none",
    "&:hover": {
      color: "#dc004e",
    },
  },
});

const BottomNavbar = (props) => {
  const classes = useStyles();
  return (
    <footer>
      <BottomNavbarStyle>
        <BottomNavbarUlStyle>
          <BottomNavbarLiStyle>
            <AppsIcon />
            <Link to="/feed" className={classes.links}>
              Feed
            </Link>
          </BottomNavbarLiStyle>
          <BottomNavbarLiStyle>
            <PersonIcon />
            <Link to="/profile" className={classes.links}>
              Profile
            </Link>
          </BottomNavbarLiStyle>
          <BottomNavbarLiStyle>
            <NotificationsIcon />
            <Link to="/notifications" className={classes.links}>
              Notifications
            </Link>
          </BottomNavbarLiStyle>
          <BottomNavbarLiStyle>
            <ExploreIcon />
            <Link to="/explore" className={classes.links}>
              Explore
            </Link>
          </BottomNavbarLiStyle>
        </BottomNavbarUlStyle>
      </BottomNavbarStyle>
    </footer>
  );
};

export default BottomNavbar;
