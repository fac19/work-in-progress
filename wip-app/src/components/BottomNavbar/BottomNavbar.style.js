import styled from "styled-components";

const BottomNavbarStyle = styled.nav`
  height: 4rem;
  border: 1px solid black;
  position: fixed;
  background-color: white;
  width: 100%;
  bottom: 0;
`;

const BottomNavbarUlStyle = styled.ul`
  display: flex;
  flex-direction: row;
  list-style: none;
  justify-content: space-evenly;
  padding: 0;
`;

export { BottomNavbarStyle, BottomNavbarUlStyle };
