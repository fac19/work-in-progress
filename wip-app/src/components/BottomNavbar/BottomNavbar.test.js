import React from "react";
import TestRenderer from "react-test-renderer";
import BottomNavbar from "./BottomNavbar";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

describe("Snapshot test", () => {
  const history = createMemoryHistory();
  it("renders the correct ui", () => {
    const component = TestRenderer.create(
      <Router history={history}>
        <BottomNavbar />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
