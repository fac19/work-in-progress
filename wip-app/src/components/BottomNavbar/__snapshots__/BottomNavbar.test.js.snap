// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<footer>
  <nav
    className="sc-AxjAm iacFcm"
  >
    <ul
      className="sc-AxirZ dKUKhO"
    >
      <li
        className="sc-AxiKw kzAwcO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
          />
        </svg>
        <a
          className="makeStyles-links-1"
          href="/feed"
          onClick={[Function]}
        >
          Feed
        </a>
      </li>
      <li
        className="sc-AxiKw kzAwcO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <a
          className="makeStyles-links-1"
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
      <li
        className="sc-AxiKw kzAwcO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
          />
        </svg>
        <a
          className="makeStyles-links-1"
          href="/notifications"
          onClick={[Function]}
        >
          Notifications
        </a>
      </li>
      <li
        className="sc-AxiKw kzAwcO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
          />
        </svg>
        <a
          className="makeStyles-links-1"
          href="/explore"
          onClick={[Function]}
        >
          Explore
        </a>
      </li>
    </ul>
  </nav>
</footer>
`;
